### PR TITLE
Adjusted behavior, default values and naming of MergingParameters

### PR DESCRIPTION
**Pull request by uuqjz**
This PR does 3 things:

1. Change mergeBuffer from being relative to MTM to an absolute Value
2. Set sensible default values for mergeBuffer and seperatingThreshold
3. Rename mergeBuffer to neighborLength and separatingThreshold to gapSize

### CODE REVIEW

File: cli/src/main/java/de/jplag/cli/CliOptions.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public class CliOptions implements Runnable {
     @ArgGroup(validate = false, heading = "Clustering%n")
     public Clustering clustering = new Clustering();
 
-    @ArgGroup(validate = false, heading = "Match Merging defense mechanism against obfuscation that merges neighboring matches based on these parameters:%n")
+    @ArgGroup(validate = false, heading = "Match Merging defense that merges neighboring matches to increase the similarity between plagiarisms:%n")
```
Comment (tsaglam) [inline cli/src/main/java/de/jplag/cli/CliOptions.java:62]:
```suggestion
    @ArgGroup(validate = false, heading = "Merging of neighboring matches to increase the similarity of concealed plagiarism:%n")
```

File: cli/src/main/java/de/jplag/cli/CliOptions.java [hunk 2]
```diff
@@ -116,13 +116,11 @@ public static class Merging {
         @Option(names = {"--match-merging"}, description = "Enables match merging (default: false)%n")
         public boolean enabled;
 
-        @Option(names = {
-                "--merge-buffer"}, description = "Defines how much lower the length of a match can be than the minimum match length (default: 0)%n")
-        public int mergeBuffer;
+        @Option(names = {"--neighbor-length"}, description = "Defines how low the length of a match can be, to be considered (default: 2)%n")
```
Comment (tsaglam) [inline cli/src/main/java/de/jplag/cli/CliOptions.java:119]: How long? Or how short?

File: core/src/main/java/de/jplag/merging/MergingParameters.java [hunk 3]
```diff
@@ -2,17 +2,17 @@
...
-public record MergingParameters(boolean enabled, int mergeBuffer, int seperatingThreshold) {
+public record MergingParameters(boolean enabled, int neighborLength, int gapSize) {
 
     /**
-     * The default values of MergingParameters are false for the enable-switch and 0 for both mergeBuffer and
-     * seperatingThreshold. These completely deactivate MatchMerging.
+     * The default values of MergingParameters are false for the enable-switch, which deactivate MatchMerging, while
+     * neighborLength and gapSize default to (2,6), which in testing yielded the best results.
      */
     public MergingParameters() {
```
Comment (tsaglam) [inline core/src/main/java/de/jplag/merging/MergingParameters.java:14]: `MergingOptions` would be more in lien with the `ClusteringOptions`.

File: core/src/main/java/de/jplag/merging/MergingParameters.java [hunk 4]
```diff
@@ -2,17 +2,17 @@
 
 /**
  * Collection of parameters that describe how a match merging should be performed.
- * @param mergeBuffer describes how shorter a match can be than the Minimum Token Match (Defaults to 0).
- * @param seperatingThreshold describes how many tokens can be between to neighboring matches (Defaults to 0).
+ * @param neighborLength describes how short a match can be, to be considered (Defaults to 2).
+ * @param gapSize describes how many tokens can be between to neighboring matches (Defaults to 6).
  */
-public record MergingParameters(boolean enabled, int mergeBuffer, int seperatingThreshold) {
+public record MergingParameters(boolean enabled, int neighborLength, int gapSize) {
```
Comment (tsaglam) [inline core/src/main/java/de/jplag/merging/MergingParameters.java:8]: `minimumNeighborLength` would match the naming scheme of the `minimumTokenMatch`.

**Review by tsaglam — CHANGES_REQUESTED** [review]
Just a few thoughts regarding the naming.

File: core/src/main/java/de/jplag/merging/MergingParameters.java (hunk 4 above)
Comment (uuqjz) [reply core/src/main/java/de/jplag/merging/MergingParameters.java:8]: What about the cli option?
Wont `--minimum-neighbor-length` be too long

File: cli/src/main/java/de/jplag/cli/CliOptions.java (hunk 2 above)
Comment (uuqjz) [reply cli/src/main/java/de/jplag/cli/CliOptions.java:119]: short would be better

File: core/src/main/java/de/jplag/merging/MergingParameters.java (hunk 4 above)
Comment (uuqjz) [reply core/src/main/java/de/jplag/merging/MergingParameters.java:8]: Following the same logic, `gapSize` should be `maximumGapSize`, which would get tedious to write.
Ultimately it is your decision.

File: core/src/main/java/de/jplag/merging/MergingParameters.java (hunk 3 above)
Comment (uuqjz) [reply core/src/main/java/de/jplag/merging/MergingParameters.java:14]: Do you know an efficient way to rename the file, class and all references in eclipse?

File: core/src/main/java/de/jplag/merging/MergingParameters.java (hunk 4 above)
Comment (uuqjz) [reply core/src/main/java/de/jplag/merging/MergingParameters.java:8]: So `maximumGapSize`, `--maximum-gap-size` `minimumNeighborLength`and `--minimum-neighbor-length` ?

Comment (tsaglam) [reply core/src/main/java/de/jplag/merging/MergingParameters.java]: for the CLI flags, we could leave minimum and maximum away, as we can add it directly to the flag description. They are more often written than read.

**Review by tsaglam — APPROVED** [review]

**Pull request merged**